### PR TITLE
Add jointcal metrics and specifications tickets/DM-9542

### PR DESCRIPTION
**Pull request by parejkoj**
Add initial jointcal metrics, and some specifications to allow testing them.

### CODE REVIEW

File: metrics/jointcal.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# validate_drp metric definitions
```
Comment (jonathansick) [inline metrics/jointcal.yaml]: Probs want to change this comment.

File: specs/jointcal/README.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+##################
+specs/jointcal
+##################
```
Comment (jonathansick) [inline specs/jointcal/README.rst]: Make sure the under/overline are the same length as the text.

File: specs/jointcal/hsc_r.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+---
+id: 'base'
+provenance_query:
+  # Prototype for query provenance. This needs to be figured out.
+  dataset_repo_url: 'https://github.com/lsst/testdata_jointcal.git'
+
+---
+# jointcal.astrometry_ndof.hsc_11_visits
+name: "hsc_11_visits"
+  base: ["#astrometry_ndof-base", "#base"]
```
Comment (jonathansick) [inline specs/jointcal/hsc_r.yaml]: I think the syntax here will be `'astrometry_ndof.yaml#astrometry_ndof-base'` to help resolve that partial.

Comment (parejkoj) [inline specs/jointcal/hsc_r.yaml]: Thanks. I've cleaned that up, and touched up the rest of this file too. I think these specifications are going to have to remain placeholders until we have Specification working, with a linter.

**Pull request merged**